Index: lib/Sema/SemaDeclAttr.cpp
===================================================================
--- lib/Sema/SemaDeclAttr.cpp	(revision 319532)
+++ lib/Sema/SemaDeclAttr.cpp	(working copy)
@@ -5997,6 +5997,9 @@
     S.Diag(Attr.getLoc(), diag::err_stmt_attribute_invalid_on_decl)
         << Attr.getName() << D->getLocation();
     break;
+  case AttributeList::AT_LoopFlatten:
+    S.Diag(Attr.getLoc(), diag::err_attribute_invalid_on_decl) << Attr.getName() << D->getLocStart();
+    break;
   case AttributeList::AT_Interrupt:
     handleInterruptAttr(S, D, Attr);
     break;
